evaluators/rank/NdcgEvaluator/__init__.py

```diff
@@ -0,0 +1,55 @@
+from math import log
+from typing import Sequence, Any
+
+from jina.executors.evaluators.rank import BaseRankingEvaluator
+
+
+def _compute_dcg(gains, power_relevance):
+    """Compute discounted cumulative gain."""
+    ret = 0.0
+    if not power_relevance:
+        for score, position in zip(gains[1:], range(2, len(gains) + 1)):
+            ret += score / log(position, 2)
+        return gains[0] + ret
+    for score, position in zip(gains, range(1, len(gains) + 1)):
+        ret += (pow(2, score) - 1) / log(position + 1, 2)
+    return ret
+
+
+def _compute_idcg(gains, power_relevance):
+    """Compute ideal discounted cumulative gain."""
+    sorted_gains = sorted(gains, reverse=True)
+    return _compute_dcg(sorted_gains, power_relevance)
+
+class NDCGEvaluator(BaseRankingEvaluator):
+    """
+    :class:`NDCGEvaluator` evaluates normalized discounted cumulative gain for information retrieval.
+    """
+
+    @property
+    def metric(self):
+        return f'nDCG@{self.eval_at}'
+
+    def evaluate(
+            self,
+            actual: Sequence[Any],
+            desired: Sequence[Any],
+            power_relevance=True,
```

Comment:
I did not see it, `power_relevance` should be a parameter of the class but not something in `evaluate` signature.

This class will be driven by a `driver` that will just consider the `BaseEvaluator` interface and will never provide an extra parameter for this.

Reply:
the signature is not real, if I pass `str` it won't be able to power them, what about `Sequence[float]` or `Sequence[Union[int, float]]`?